crates/sui-framework/sources/safe.move

```diff
@@ -0,0 +1,214 @@
+/// The Safe standard is a minimalistic shared wrapper around a coin. It provides a way for users to provide third-party dApps with
+/// the capability to transfer coins away from their wallets, if they are provided with the correct permission.
+module sui::safe {
+    use sui::object::{Self, ID, UID};
+    use sui::tx_context::{TxContext, sender};
+    use sui::transfer::Self;
+    use sui::balance::{Self, Balance};
+    use std::vector::Self;
+
+    ///
+    /// Allows any holder of a capability to transfer a fixed amount of assets from the safe.
+    /// Useful in situations like an NFT marketplace where you wish to buy the NFTs at a specific price.
+    /// 
+    /// @ownership: Shared
+    /// 
+    struct Safe<phantom T> has key {
+        id: UID,
+        owner: address,
+        balance: Balance<T>,
+        permissions_bitmap: vector<u64>,
```

Comment:
https://github.com/MystenLabs/sui/blob/main/crates/sui-framework/deps/move-stdlib/sources/bit_vector.move might be useful here--I think it is desirable to separate the (rather generic) implementation of the bitvector data structure from the (rather specific) implementation of `Safe`.

Reply:
I was looking at the BitVector implementation, and was wondering whether it could potentially be less efficient than just storing u64s? First observation is that it stores a vector of bools, which I presume is a byte each rather than a bit (you probably know best). But my biggest concern is that it has a fixed size at instantiation, where as the variance for usage on each safe is likely huge. I'll change to BitVector for now, but I'll keep the old commit unsquashed incase.